File: src/my_class_pkg/src/ros_server.cpp
#include "ros/ros.h"
#include "my_class_pkg/MyServiceMsg.h"

bool serviceCallback(my_class_pkg::MyServiceMsg::Request &req,
                     my_class_pkg::MyServiceMsg::Response &res) {
    res.output = req.input * 2;
    ROS_INFO("Request received: input = %ld, output = %ld", (long int)req.input, (long int)res.output);
    return true;
}

int main(int argc, char **argv) {
    ros::init(argc, argv, "service_server");
    ros::NodeHandle nh;
    ros::ServiceServer server = nh.advertiseService("my_service", serviceCallback);
    ROS_INFO("Service server started, waiting for requests...");
    ros::spin();
    return 0;
}

File: src/my_class_pkg/scripts/ros_server.py
#!/usr/bin/env python3
import rospy
from my_class_pkg.srv import MyServiceMsg, MyServiceMsgResponse

# Service callback function: handle request and return response
def handle_service_request(req):
    rospy.loginfo(f"Request received: input = {req.input}")
    return MyServiceMsgResponse(req.input * 2)

if __name__ == "__main__":
    # Initialize ROS node
    rospy.init_node("python_service_server")
    # Advertise service (service name: my_service)
    rospy.Service("my_service", MyServiceMsg, handle_service_request)
    rospy.loginfo("Python Service Server started, waiting for requests...")
    # Keep server running
    rospy.spin()

